runtime-modules/project-token/src/tests/bonding_curve.rs

```diff
@@ -0,0 +1,1079 @@
+        let token = Token::token_info_by_id(token_id);
+        assert!(token.bonding_curve.is_none());
+    })
+}
+
+#[test]
+fn deactivate_ok_with_full_cycle_from_activation() {
+    let token_id = token!(1);
+    let ((user_member_id, user_account_id), user_balance) = (member!(2), joy!(5_000_000));
+    build_default_test_externalities_with_balances(vec![(user_account_id, user_balance)])
+        .execute_with(|| {
+            IssueTokenFixture::default().execute_call().unwrap();
+            ActivateAmmFixture::default()
+                .with_creator_reward(Permill::zero())
+                .execute_call()
+                .unwrap();
+            BondFixture::default()
+                .with_sender(user_account_id)
+                .with_member_id(user_member_id)
+                .with_amount(DEFAULT_BONDING_AMOUNT)
+                .execute_call()
+                .unwrap();
+
+            UnbondFixture::default()
+                .with_sender(user_account_id)
+                .with_member_id(user_member_id)
+                .with_amount(DEFAULT_BONDING_AMOUNT)
+                .execute_call()
+                .unwrap();
+
+            DeactivateAmmFixture::default().execute_call().unwrap();
+
+            let amm_treasury_account = Token::module_bonding_curve_reserve_account(token_id);
+            assert_eq!(
+                Balances::usable_balance(amm_treasury_account),
+                ExistentialDeposit::get()
+            );
+        })
+}
+
+#[test]
+fn amm_deactivation_ok_with_event_deposit() {
+    let token_id = token!(1);
+    let (creator_id, creator_account) = member!(1);
+    let config = GenesisConfigBuilder::new_empty().build();
+    build_test_externalities(config).execute_with(|| {
+        IssueTokenFixture::default().execute_call().unwrap();
+        ActivateAmmFixture::default()
+            .with_token_id(token_id)
+            .with_sender(creator_account)
+            .with_member_id(creator_id)
+            .execute_call()
+            .unwrap();
+        DeactivateAmmFixture::default()
+            .with_token_id(token_id)
+            .with_sender(creator_account)
+            .with_member_id(creator_id)
+            .execute_call()
+            .unwrap();
+
+        last_event_eq!(RawEvent::BondingCurveDeactivated(token_id, creator_id,));
+    })
+}
+
+#[test]
+fn token_state_idle_after_amm_deactivation() {
+    let token_id = token!(1);
+    let config = GenesisConfigBuilder::new_empty().build();
+    build_test_externalities(config).execute_with(|| {
+        IssueTokenFixture::default().execute_call().unwrap();
+        ActivateAmmFixture::default().execute_call().unwrap();
+        DeactivateAmmFixture::default().execute_call().unwrap();
+
+        let token = Token::token_info_by_id(token_id);
+        assert_eq!(IssuanceState::of::<Test>(&token), IssuanceState::Idle);
+    })
+}
+
+// --------------------------------- EXTRA ----------------------------------------
+
+#[test]
+fn review_eval_function() {
```

Comment:
Also, I think you should use some non-zero `b` for this test, otherwise it seems kind of incomplete